test/apps/iodemo/io_demo.cc

```diff
@@ -1006,18 +997,17 @@ class DemoServer : public P2pDemoCommon {
         ConnectionStat &conn_stat  = _conn_stat_map.find(conn->id())->second;
 
         iov->init(msg->data_size, _data_chunks_pool, msg->sn, opts().validate);
-        w->init(this, conn, msg->sn, iov,
-                conn_stat.get_completion_counter<IO_WRITE>());
+        w->init(this, conn, msg->sn, iov, &conn_stat.completions<IO_WRITE>());
         assert(iov->size() == 1);
 
-        conn_stat.op_started<IO_WRITE>(msg->data_size);
+        conn_stat.bytes<IO_WRITE>() += msg->data_size;
         conn->recv_am_data((*iov)[0].buffer(), (*iov)[0].size(), data_desc, w);
     }
 
     virtual void dispatch_connection_accepted(UcxConnection* conn) {
         bool res = _conn_stat_map.insert(
                 std::make_pair(conn->id(), ConnectionStat())).second;
-        assert(res);
+        assert(res && "connection duplicate in statistics map");
```

Comment:
looks weird, i's suggest to print error with LOG (so it will include hostname etc) and then call abort